src/string.cr

```diff
@@ -2302,7 +2304,7 @@ class String
   # ```
   # "aabbcc".delete { |c| ['a', 'b'].includes?(c) } # => "cc"
   # ```
-  def delete
+  def delete(&block : Char -> Bool) : String
```

Comment:
Sometimes you have typed these blocks as `-> _`, so that truthiness works, sometimes you typed as `-> Bool` so you need to explicitly return a bool.

The former is descriptive, the latter is prescriptive, a breaking change, and belongs in another PR if at all.